app/src/main/java/com/android/unio/model/events/Event.kt

```diff
@@ -0,0 +1,25 @@
+package com.android.unio.model.events
+
+/**
+ * Represents an event with detailed information.
+ *
+ * @property id The unique identifier for the event.
+ * @property title The title of the event.
+ * @property catchy_description A short, attention-grabbing description of the event.
+ * @property description A detailed description of the event.
+ * @property date The date of the event in string format. Consider using a proper Date type for
+ *   better handling of date values.
+ * @property location The location where the event will take place.
+ * @property main_type The main category or type of the event (e.g., conference, concert, etc.).
+ * @property picture A URL or path to the event's main picture or image.
+ */
+data class Event(
+    val id: String,
+    val title: String,
+    val catchy_description: String,
+    val description: String,
+    val date: String, // Consider using Date type with proper formatting
+    val location: String,
```

Comment:
> The "location" attribute should be a "Location" type, like used in the bootcamp, or another type, from a new class, to fit better with EPFL geography.

Yes. We do have a Location data class available which can and should be used

Reply:
Agreed, I believe Firebase provides a GeoPoint data class we could use.

Reply:
same as before, i just did it blindly but i will destroy this class for the better one

Reply:
> Agreed, I believe Firebase provides a GeoPoint data class we could use.

True, however this would only store the coordinates correct?

Reply:
You can transform Coordinates into a place with Google Maps API 
I think the best thing to do would be to ask the Event creator precise coordinates (as our map feature is based on precision of the event) and then we could show himthe output of Maps API for example 'Plage de Vidy' 
Then he can agree to it or edit it, and the location as a text would be what people see on the event page, but if they clic on the map the pin point location would be the precise coordinates of the event

Reply:
Yes correct I just mean we will still need to hava a `String` field for the name of the location, so we either keep `Location.kt` as is, or use the `GeoPoint` data class instead of having two `Double` fields

Reply:
> True, however this would only store the coordinates correct?

Yes, the `GeoPoint` class contains only a longitude and latitude property, but its advantage is the automatic serialization and hydration when sending/retrieving from Firebase. Maybe the correct way to go is a wrapper class that contains a `GeoPoint` in addition to some information about the event's location.